### PR TITLE
[MM-57010] Include client type in websocket connections metric

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR extends the new `origin_client` label to the `mattermost_http_websockets_total` gauge so that it will be easy to see the actual distribution of the different clients we support.

I opted not to add a new metric but extend the existing one which is mostly a backwards compatible change although the Grafana panel will need to be updated to make sense of this (see screenshots below). Let me know if you have concerns.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-57010

#### Screenshots

*Original panel*

image

*Updated panel*

image

#### Release Note

```release-note
Added an `origin_client` label to the `mattermost_http_websockets_total` Prometheus metric.
```

### CODE REVIEW

**Review by agarciamontoro — APPROVED** [review]
Looks good to me. Should we address the dashboard changes once this is in and we can see what's broken? Maybe look at the V2 dashboard first?

**Comment by streamer45** [comment]
> Looks good to me. Should we address the dashboard changes once this is in and we can see what's broken? Maybe look at the V2 dashboard first?

Nothing should break but connections would be split without labels so it can get confusing (see first screenshot). As soon as this is approved I'll port the changes over to https://github.com/mattermost/mattermost-performance-assets and update the load-test side as well.

File: server/channels/app/platform/web_conn.go [hunk 1]
```diff
@@ -88,6 +89,7 @@ type WebConn struct {
 	Locale           string
 	Sequence         int64
 	UserId           string
+	OriginClient     string
```
Comment (agnivade) [inline server/channels/app/platform/web_conn.go:92]: Does this need to be exported?

**Review by agnivade — COMMENTED** [review]
We also need to update the v2 dashboard in grafana.com. I have the credentials to the site. Once you update the JSON in the performance assets repo, we can upload it to grafana.com

File: server/channels/app/platform/web_conn.go (hunk 1 above)
Comment (streamer45) [reply server/channels/app/platform/web_conn.go:92]: Not really, we are using the config to get it there and then it's used internally. I can update :+1:

**Review by agnivade — APPROVED** [review]

**Pull request merged**